### DISCUSSION TITLE
Installation Issue (pcapErrorNotActivated)

### DISCUSSION THREAD

**Discussion by noob0x01** [projectdiscovery/naabu#694] [Q&A] [answered]
while trying install latest version facing this error :

# github.com/google/gopacket/pcap
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:30:22: undefined: pcapErrorNotActivated
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:52:17: undefined: pcapTPtr
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:64:10: undefined: pcapPkthdr
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:103:6: undefined: pcapBpfProgram
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:110:7: undefined: pcapPkthdr
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:268:33: undefined: pcapErrorActivated
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:269:33: undefined: pcapWarningPromisc
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:270:33: undefined: pcapErrorNoSuchDevice
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:271:33: undefined: pcapErrorDenied
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:748:14: undefined: pcapTPtr
go/pkg/mod/github.com/google/gopacket@v1.1.19/pcap/pcap.go:271:33: too many errors

note : libpcap-dev installed .

![Screenshot 2023-05-05 at 11 57 58 PM](https://user-images.githubusercontent.com/51519007/236533627-e54e955f-e078-483b-9168-d5c5d004afaa.png)

**Comment by ehsandeep** [comment]
@noob0x01 see 
- https://github.com/projectdiscovery/naabu#prerequisite
- https://github.com/projectdiscovery/naabu/discussions/139

**Comment by noob0x01** [comment]
Hey @ehsandeep 

Not working. 
![Screenshot 2023-05-06 at 12 44 17 AM](https://user-images.githubusercontent.com/51519007/236542728-60928572-06e2-42fe-92e5-35f122e097bd.png)
I think this one is a different error.

**Comment by ehsandeep** [comment]
@noob0x01 can you share the output of `naabu -hc`?

**Comment by noob0x01** [comment]
![Screenshot 2023-05-06 at 12 48 12 AM](https://user-images.githubusercontent.com/51519007/236543340-e9ea75b5-c291-4e32-8c1d-eb70d892e99d.png)
@ehsandeep

**Comment by ehsandeep** [comment]
@noob0x01, what's the go version you have installed? `go version`?

**Comment by noob0x01** [comment]
Hey, @ehsandeep Latest one. (go version go1.20 linux/amd64)
![Screenshot 2023-05-06 at 12 51 09 AM](https://user-images.githubusercontent.com/51519007/236543869-429ffc6d-2b13-4dfd-93f2-355e6107c487.png)

**Comment by UxplnJDK** [comment]
i was also facing the same installation issue using installation link..i just downloaded binary release and executed it..binary works fine.

**Comment by sam5epi0l** [comment]
You might want to take a look at these similar issues:

- google/gopacket#1018
- google/gopacket#629

It looks like this error is caused due to go env variables. Try setting `CGO_ENABLED=1`

**Comment by noob0x01** [comment]
Hey @all 

- If you using latest go version kindly downgrade(remove and reinstall go 1.19 )  `go version go1.19.4 linux/amd64`

- second if you faced `compiler "gcc" not found: exec: "gcc": executable file not found in $PATH`

- `sudo apt-get install build-essential`

This solved the problem. It installs the gcc/g++ compilers and libraries.

[a comment by an automated account is not shown]

**Reply by boobooHQ** [reply]
Works with these commands . reinstall naabu after running these commands

**Reply by 0xcrypto** [reply]
In fedora, install gcc and libpcap-devel.
